game/lib/widget/leaderboard_cell.dart

```diff
@@ -78,7 +85,8 @@ Widget leaderBoardCell(
                       padding: const EdgeInsets.only(left: 10, right: 5),
                       child: Align(
                         alignment: Alignment.centerLeft,
-                        child: Text(name, style: nameStyle),
+                        child: Text(truncateWithEllipsis(name, 10),
```

Comment:
This probably works fine, but I do think that it would be better to use Flutter's built in Text overflow handler with "overflow: TextOverflow.ellipsis". You might need to add some wrapper widgets though.